### PR TITLE
Al/inspectors

### PR DESCRIPTION
**Pull request by alundgard**
Refactor /inspectors. Add prop interfaces for state and dispatch.

### CODE REVIEW

File: src/js/components/inspectors/SpatialPreset.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  band: boolean,
+  group: boolean,
+  scale: any // TODO: propTypes.instanceOf(Immutable.Map),
+  name: string,
+  setPreset: any,
+  reset: any,
+  className: any
+}
+
+class BaseSpatialPreset extends React.Component<SpatialPresetProps>{
```
Comment (jonathanzong) [inline src/js/components/inspectors/SpatialPreset.tsx]: `SpatialPresetProps` is currently redundant with `OwnProps`, `StateProps`, and `DispatchProps`.

Just as a reminder: the generic type `P` in `React.Component<P>` represents the prop types for the class. The props are split into `OwnProps`, which are regular props; `StateProps`, which are the outputs of mapStateToProps; and `DispatchProps`, which are the outputs of mapDispatchToProps. So in this case you want to:

1. Change the type in this line from `SpatialPresetProps` to `OwnProps & StateProps & DispatchProps`
2. Move fields that are present on `SpatialPresetProps` but not present in any of `OwnProps`, `StateProps`, and `DispatchProps` to `OwnProps`
3. Delete the definition for `SpatialPresetProps`

Please double check the other files for instances of this too :)

File: src/js/components/inspectors/Text.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        'properties.update.text.template', val));
+    }
+  };
+}
+
+interface TextInspectorProps {
+  primId: number,
+  primType: any, // TODO
+  autoVal: string,
+  dsId: any,
```
Comment (jonathanzong) [inline src/js/components/inspectors/Text.tsx]: `dsId` is a `number`

(it's usually referring to a dataset id)

**Review by jonathanzong — CHANGES_REQUESTED** [review]
nice work! one class of changes to make and components should be good to go 🎉

File: src/js/components/inspectors/SpatialPreset.tsx (hunk 1 above)
Comment (alundgard) [reply src/js/components/inspectors/SpatialPreset.tsx]: 👍

File: src/js/components/inspectors/Text.tsx (hunk 2 above)
Comment (alundgard) [reply src/js/components/inspectors/Text.tsx]: 👍

**Review by jonathanzong — APPROVED** [review]

**Pull request merged**